fails:Kernel.throw raises NameError if there is no catch block for the symbol
fails:Kernel.throw raises TypeError if the first argument is not a symbol
